### DISCUSSION TITLE
How to grant repo clone to Reusable workflows without a PERSONAL access token?

### DISCUSSION THREAD

**Discussion by sammcj** [community/community#31423] [Actions]
We make extensive use of reusable workflows and composite actions.

For these - they often need to check out repositories such as the repo they're being called from, their own repo and at times a shared scripts repo.

We are required to remove all **PERSONAL** access tokens (PATs) for security, compliance and best practice, however we can't see a way to grant access for reusable workflows to clone other repositories without an individual's personal access token.

Github's documentation instructs customers to use _personal_ access tokens for workflows/actions - this seems dangerous and counter intuitive.

Options I've thought of but don't like the sound of:

1. Create fake / bot user accounts that cost us a license and use their PATs (Which seems dangerous and in breach of TOS).
2. Use some sort of Github App to auth and get a temporary token to clone repos every workflow run (Which seems costly, adds complexity and creates technical debt).

It also seems that [there is no scope for PATs to have read only access to clone repos](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes) - it's full access or nothing!

- What is the correct way of handling reusable workflow access to a set of repositories without a _personal_ access token at present?

**Comment by andres701** [comment]
totalmente todo o nada ja

**Reply by sammcj** [reply]
Sorry I don't understand?
